examples/next-minimal-starter/src/pages/api/trpc/[trpc].ts

```diff
@@ -4,29 +4,44 @@
  */
 import * as trpcNext from '@trpc/server/adapters/next';
 import { z } from 'zod';
-import { publicProcedure, router } from '~/server/trpc';
+import { publicProcedure, unstable_toRouter } from '~/server/trpc';
 
-const appRouter = router({
-  greeting: publicProcedure
-    // This is the input schema of your procedure
-    // 💡 Tip: Try changing this and see type errors on the client straight away
+class PostService {
+  public allPosts = publicProcedure.query(() => {
+    return [
+      {
+        id: 1,
+        title: 'hello world',
+      },
+    ];
+  });
+}
+class RootService {
+  public post;
+  constructor() {
+    this.post = unstable_toRouter(new PostService());
+  }
```

Comment:
Continuing https://github.com/trpc/trpc/pull/3727/files#r1097883151

```suggestion
  constructor(PostServiceImpl = PostService) {
    this.post = unstable_toRouter(PostService);
  }
```